api/core/tools/provider/builtin/duckduckgo/tools/ddgo_video.py

```diff
@@ -36,10 +39,29 @@ def _invoke(self, user_id: str, tool_parameters: dict[str, Any]) -> list[ToolInv
             title = res.get("title", "")
             embed_html = res.get("embed_html", "")
             description = res.get("description", "")
+            content_url = res.get("content", "")
+
+            # Handle TED.com videos
+            if not embed_html and "ted.com/talks" in content_url:
+                embed_url = content_url.replace("www.ted.com", "embed.ted.com")
+                if proxy_url:
+                    embed_url = f"{proxy_url}{embed_url}"
+                embed_html = """
+<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden; \
+max-width: 100%; border-radius: 8px;">
+    <iframe
+        style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"
+        src="{src}"
+        frameborder="0"
+        allowfullscreen>
+    </iframe>
+</div>""".format(src=embed_url)
```

Comment:
I think this part is duplicated with belows, maybe combine them can be more clear